src/core_modules/capture-core/components/WidgetProfile/DataEntry/dataEntry.epics.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+import { ofType } from 'redux-observable';
+import uuid from 'uuid/v4';
+import { map } from 'rxjs/operators';
+import { batchActions } from 'redux-batched-actions';
+import { dataEntryActionTypes, updateTei, saveSucceed, setTeiModalState, TEI_MODAL_STATE } from './dataEntry.actions';
+
+const geometryType = (key) => {
+    const types = ['Point', 'None', 'Polygon'];
+    return types.find(type => key.toLowerCase().includes(type.toLowerCase()));
+};
+
+const standardGeoJson = (geometry) => {
+    if (!geometry) {
+        return undefined;
+    }
+    if (Array.isArray(geometry)) {
+        return {
+            type: 'Polygon',
+            coordinates: geometry,
+        };
+    } else if (geometry.longitude && geometry.latitude) {
+        return {
+            type: 'Point',
+            coordinates: [geometry.longitude, geometry.latitude],
+        };
+    }
+    return undefined;
+};
+
+const deriveAttributesFromFormValues = (formValues = {}) =>
+    Object.keys(formValues)
+        .filter(key => !geometryType(key))
+        .map(key => ({ attribute: key, value: formValues[key] }));
+
+const deriveGeometryFromFormValues = (formValues = {}) =>
+    Object.keys(formValues)
+        .filter(key => geometryType(key))
+        .reduce((acc, currentKey) => standardGeoJson(formValues[currentKey]), undefined);
+
+export const updateTeiEpic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(dataEntryActionTypes.TEI_UPDATE_REQUEST),
+        map((action) => {
+            const uid = uuid();
+            const { formsValues } = store.value;
+            const {
+                dataEntryId,
+                itemId,
+                orgUnitId,
+                trackedEntityTypeId,
+                trackedEntityInstanceId,
+                onSaveExternal,
+                onSaveSuccessActionType,
+                onSaveErrorActionType,
+            } = action.payload;
+
+            const values = formsValues[`${dataEntryId}-${itemId}`];
+            const serverData = {
+                attributes: deriveAttributesFromFormValues(values),
```

Comment:
This is the same code we use when we create a new TEI. The values are not properly converted for server format. There is [another PR opened](https://github.com/dhis2/capture-app/pull/2267) to fix this for new TEI. Once that PR is merged I will update the code here also.